tests/orm/test_querybuilder.py

```diff
@@ -852,6 +852,76 @@ def test_empty_filters(self):
         qb = orm.QueryBuilder().append(orm.Data, filters={'or': [{}, {}]})
         assert qb.count() == count
 
+    @pytest.mark.usefixtures('suppress_internal_deprecations')
+    @pytest.mark.usefixtures('aiida_profile_clean')
+    def test_abstract_code_filtering(self, aiida_localhost, aiida_code, tmp_path):
+        """Test that querying for AbstractCode correctly returns all code instances.
+
+        This tests the fix for issue #6687, where QueryBuilder couldn't find codes
+        when looking for AbstractCode due to a node_type mismatch.
+        """
+        installed_code = aiida_code(
+            'core.code.installed',
+            label='installed-code',
+            computer=aiida_localhost,
+            filepath_executable='/bin/bash',
+        )
```

Comment:
I like how using the fixture ends up with the same amount of code.

Reply:
for the `installed_code` we could you the `aiida_code_installed` fixture, but I thought I'd keep things symmetric for all code types is better.